lib/mongoid/association/embedded/batchable.rb

```diff
@@ -232,7 +237,11 @@ def normalize_docs(docs)
         #
         # @return [ String ] The atomic path.
         def path
-          @path ||= _unscoped.first.atomic_path
+          unless _unscoped.empty?
+            @path ||= _unscoped.first.atomic_path
+          else
```

Comment:
It is generally not recommended to use `unless` with `else` clause - https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/UnlessElse
I think it would be more readable with `if`.

Reply:
:+1:

Reply:
Additionally I find it more readable to perform the assignment once, i.e.:

```
@path ||= if ...
  ...
else
  ...
end
```

Reply:
done